src/js_native_api_types.h

```diff
@@ -108,6 +108,37 @@ typedef enum {
 //   * the definition of `napi_status` in doc/api/n-api.md to reflect the newly
 //     added value(s).
 
+#ifdef NAPI_EXPERIMENTAL
+// Features allow changing internal behavior of existing Node-API functions.
+//
+// We pass a node_api_features pointer to the napi_module struct
+// in the NAPI_MODULE_X macro. This macro is used for the module registration.
+// If the module is initialized without using this macro, then there will be
+// no features selected and the module will use the node_api_feature_none.
+//
+// Each Node-API version defines its own default set of features.
+// For the current version it can be accessed using node_api_default_features.
+// A module can override the set of its enabled features by adding
+// NODE_API_CUSTOM_FEATURES definition to the .gyp file and then defining the
+// value of the global node_api_module_features variable.
+// To check enabled features use the `node_api_is_feature_enabled` function.
+//
+// For example, to disables node_api_feature_reference_all_types:
+// node_api_features node_api_module_features =
+//   node_api_default_features & ~node_api_feature_reference_all_types;
+typedef enum {
+  // To be used when no features needs to be set.
+  node_api_feature_none = 0,
+  // Use napi_ref for all value types.
+  // Not only objects, external objects, functions, and symbols as before.
+  node_api_feature_reference_all_types = 1 << 0,
+  // Each version of Node-API is going to have its own default set of features.
+  node_api_default_experimental_features = node_api_feature_reference_all_types,
```

Comment:
I'm not sure if we should have this, I think we should guard the availabilty to be able to turn on experimental features with NAPI_EXPERIMENTAL, but I'm not sure if having a set different than the default for the Node-API version makes sense. Otherwise somebody who wants to use any experimental API is also forced into to new defaults.

Reply:
In case if someone does not want to use the new experimental default, they can always override it per module.